resources/definitions/fdmprinter.def.json

```
@@ -1843,6 +1843,51 @@
                     "limit_to_extruder": "top_bottom_extruder_nr",
                     "settable_per_mesh": true
                 },
+                "wall_line_count_skin":
+                {
+                    "label": "Wall Line Count Top/Initial",
+                    "description": "The number of walls in the top-skin and initial-layer-skin parts. This allows for setting less (or more) walls to parts of the model 'exposed to air' (except for bottom parts higher up in the model).",
+                    "default_value": 1,
+                    "value": "wall_line_count",
+                    "minimum_value": "0",
+                    "maximum_value_warning": "10",
+                    "maximum_value": "999999",
+                    "type": "int",
+                    "enabled": "(top_layers > 0 or bottom_layers > 0)",
+                    "limit_to_extruder": "top_bottom_extruder_nr",
+                    "settable_per_mesh": true,
+                    "children":
+                    {
+                        "wall_line_count_0":
```

Is this for the initial layer (on the build plate), or the roofing layers ? I think it it the latter, then I suggest:
```suggestion
                       "wall_line_count_flooring":
```

---

Oh right, the `0` postfix is mostly for the outer-wall, so this would work better.

---

... except the flooring doesn't necessarily have to be on the bottom, I think _initial might cover it more closely (which also communicates that it's really only 1 layer).

---

Hmm, calling it `layer_0` does look confusing to me, it seems like it is only for the `initial` layer, the one on the build plate. `flooring` is definitely not perfect, but this is what we used for other settings, and also for consistency with the `roofing` settings. I let it to your judgment.

---

Yeah it _is_ only used for the initial layer, applying it to flooring in general would cause bridging to fail (and not be much better even if bridging is off -- the new 'floating' walls would still fall off the sides).

---

(See line 140 in layerPart.cpp in the engine : `layer_nr == 0 ? getTopOrBottom(-1, "wall_line_count_layer_0", layer_nr, slicer->layers, mesh.settings) : Shape(),` )